### PR TITLE
Switch DISMF to Paolo et al. (2023) dataset

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] User's Guide has been updated
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I ran `files_for_e3sm/remap_ice_shelf_melt` step and saw:
```
Rerouting fluxes from 1054 cells
Captured flux (kg/s):         18653044.8
Rerouted flux (kg/s):         11399309.2
Total flux (kg/s):            30052354.0
Total after rerouting (kg/s): 30052354.0
```
This corresponds to a total flux of 948 Gt/yr, which is pretty close to the total of 961 Gt/yr that I calculated based on the bottom row of Table 3 of the Paolo et al. (2023) paper.

**Comment by xylar** [comment]
@darincomeau and @cbegeman, this is a follow-up on #777

**Comment by cbegeman** [comment]
I support switching to this dataset. Based on this figure from Paolo et al. (2023), the Paolo dataset is more accurate than the Adusumilli dataset (panel d shows better agreement with flux gate method vs. panel c).
image
> Figure 8Comparison of estimates from this study with previous work. Comparison of (a, c) Adusumilli et al. (2020) and (b, d) this study's ice shelf basal melt estimates against a control-volume calculation of ice shelf mass change (line). The control volume is based on the input and output fluxes across the grounding line and ice front, mass gained or lost due to surface mass balance (a, b), and ice loss due to anomalies in basal melt (c, d). For this comparison, we only considered grid cells that are at least 90 % hydrostatically compensated (near fully floating).

**Comment by xylar** [comment]
With the latest changes, when I test `remap_ice_shelf_melt` on the IcoswISC30E3r5 mesh, I'm seeing:
```
min projected area ratio: 0.9547266742733075
max projected area ratio: 1.1524603930432318

Area of a cell (m^2):            3686400.0
Total flux on plane (kg/s):      30023072.2
Total flux on sphere (kg/s):     30023072.2

min MPAS area ratio: 0.9999999298969185
max MPAS area ratio: 1.0000000668051465

Total flux w/ MPAS area (kg/s):  30023072.2
Total flux w/ sphere area(kg/s): 30023072.2
Rerouting fluxes from 1054 cells
Captured flux (kg/s):            18636267.0
Rerouted flux (kg/s):            11386805.2
Total flux (kg/s):               30023072.2
Total after rerouting (kg/s):    30023072.2
```

**Comment by xylar** [comment]
I have placed new DISMF files at:
```
/lcrc/group/e3sm/public_html/inputdata/ocn/mpas-o/IcoswISC30E3r5/prescribed_ismf_paolo2023.IcoswISC30E3r5.20240221.nc
/lcrc/group/e3sm/data/inputdata/ocn/mpas-o/SOwISC12to30E3r2/prescribed_ismf_paolo2023.SOwISC12to30E3r2.20240221.nc
/lcrc/group/e3sm/data/inputdata/ocn/mpas-o/RRSwISC6to18E3r4/prescribed_ismf_paolo2023.RRSwISC6to18E3r4.20240221.nc
```

File: compass/ocean/tests/global_ocean/init/remap_ice_shelf_melt.py [hunk 1]
```diff
@@ -61,38 +72,248 @@ def run(self):
...
+
+    # convert to the units and variable names expected in MPAS-O
+
+    # Paolo et al. (2023) ice density (from attributes)
+    rho_ice = 917.
+    s_per_yr = 365. * constants['SHR_CONST_CDAY']
+    latent_heat_of_fusion = constants['SHR_CONST_LATICE']
+    ds['x'] = x
+    ds['y'] = y
+    area_ratio = planar_area / sphere_area
```
Comment (cbegeman) [inline compass/ocean/tests/global_ocean/init/remap_ice_shelf_melt.py:215]: @xylar Great!

**Comment by xylar** [comment]
@cbegeman and @darincomeau,

Could you review this and #777 based on my testing and this analysis output?
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis/20240223.v3.LR.CRYO1850-DISMF.paolo.chrysalis/

Feel free to suggest (or do) any further testing that you think is needed.

**Review by cbegeman — APPROVED** [review]
Approving based on the testing done so far, the fact that this dataset is more accurate than its predecessor, and the fact that the total ice shelf freshwater flux from the ocean analysis member agrees with the remapped dataset on the RRSwISC6to18E3r4 mesh to within 0.01 Gt/yr.

**Review by darincomeau — APPROVED** [review]
Approved based on existing testing. Thanks @xylar !

**Pull request merged**